src/Components/Admin/ObjectEditor/object.js

```diff
@@ -0,0 +1,318 @@
+                        Save
+                    </Button>
+                </div>
+            </div>
+        );
+    }
+}
+
+export default function ObjectEditor({
+    match: {
+        params: { sceneId, objectId },
+    },
+}) {
+    const classes = useStyles();
+    const handleError = useErrorHandler();
+    const [puzzleBody, setPuzzleBody] = useState({});
+    const [puzzleType, setPuzzleType] = useState("");
+    const [animationsJson, setAnimationsJson] = useState({});
+    const [isInteractable, setIsInteractable] = useState(false);
+
+    const puzzleTypeList = [
+        { value: "text-pane", label: "Text Puzzle" },
+        { value: "rotation-controls", label: "Rotation Puzzle" },
+        { value: "keypad", label: "Keypad Puzzle" },
+        { value: "visual-pane", label: "Visual Puzzle" },
+        { value: "jigsaw-puzzle", label: "Jigsaw Puzzle" },
+        { value: "ordered-puzzle", label: "Ordered Puzzle" },
+    ];
+
+    useEffect(() => {
+        const getPuzzleBody = async () => {
+            const data = await getPuzzle(sceneId, objectId, handleError);
+            setAnimationsJson(data.animations_json);
+            if (JSON.stringify(data.animations_json) === JSON.stringify({})) {
+                setPuzzleBody({ jsonData: {} });
+                setIsInteractable(false);
+            } else {
+                setPuzzleBody(data.animations_json.blackboardData);
+                setPuzzleType(
+                    data.animations_json.blackboardData.componentType
+                );
+                setIsInteractable(data.is_interactable);
+                const puzBody = data.animations_json.blackboardData;
+                if (puzBody.componentType === "text-pane") {
+                    const tempTexts = puzBody.jsonData.data;
+                    for (let i = 0; i < puzBody.jsonData.data.length; i++) {
+                        tempTexts[i]["index"] = i;
+                    }
+                    puzBody.jsonData.data = tempTexts;
+                    setPuzzleBody(puzBody);
+                }
+            }
+        };
+        if (JSON.stringify(puzzleBody) === JSON.stringify({})) {
+            getPuzzleBody();
+        }
+    }, [sceneId, objectId, puzzleBody, handleError]);
+
+    const selectPuzzleType = (obj) => {
+        if (obj) {
+            const data = puzzleBody;
+            data.componentType = obj.value;
+            if (obj.value === "text-pane") {
+                if (!data.jsonData.data) {
+                    data.jsonData.data = [];
+                }
+            }
+            setPuzzleBody(data);
+            setPuzzleType(obj.value);
+        }
+    };
+
+    const saveTexts = (texts) => {
+        const animCopy = animationsJson;
+        const puzzleBodyCopy = puzzleBody;
+        const textsCopy = texts;
+        for (let i = 0; i < textsCopy.length; i++) {
+            delete textsCopy["index"];
+        }
```

Comment:
Can you explain this logic? I have never seen the usage of `delete` in JS lol

Reply:
The index field is something I used to display the Text {index} of {texts.length} logic for the text cards, it shouldn't actually be a field sent to the backend when saving.

Reply:
Update: Reworked the logic to remove the usage of "index" as a key for texts